FirebaseSessions/Sources/Public/SessionsSubscriber.swift

```diff
@@ -28,7 +28,7 @@ public protocol SessionsSubscriber {
 /// whenever the Session changes
 @objc(FIRSessionDetails)
 public class SessionDetails: NSObject {
-  var sessionId: String?
+  @objc public var sessionId: String?
 
   public init(sessionId: String?) {
```

Comment:
@samedson Is there a way we can make `sessionId` not optional? This way when payload is passed down to Crashlytics and Fireperf, we don't have to deal with handling "nil sessionId", which I don't think should even exist at this point.

Reply:
It is a tricky situation here. Initially, the implementation @samedson had was to make SessionID non-optional (Where in case where the session ID cannot be generated, the value will be set to Empty). 

But, that had some caveats. Empty sessionID is not valid either. So, the sessionID listeners have a need to validate the accuracy of the sessionID. To handle situations where the new session ID generation has issues, making it an optional would be a easiest way to handle it for future.